From 664673245aa11277c6c698b15a5b7f6db9e9d9c5 Mon Sep 17 00:00:00 2001
From: "song.yu" <song.yu@sophgo.com>
Date: Mon, 13 Apr 2026 17:22:31 +0800
Subject: [PATCH] fix: use default partition.xml when board/xxx/partition.xml
 does not exist

[description ]:NA
[root   cause]:NA
[JIRA      ID]:CV184XSDK-1169
[chip project]:CV184X
[side effects]:NA
[Affected *.ko/*.so/*.a] :NA

Change-Id: I6151e6f86406934763dc6b5426e469f8f33d7633
---
 common_functions.sh |  2 +-
 envsetup_soc.sh     | 10 ++++++++--
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/common_functions.sh b/common_functions.sh
index f8eaaa41..fde87198 100644
--- a/common_functions.sh
+++ b/common_functions.sh
@@ -453,7 +453,7 @@ function setconfig()
 
 function _build_add_bash_completion()
 {
-  _boards=$(find "${BUILD_PATH}/boards" -mindepth 2 -maxdepth 2 -type d -not -path '*/default/*' -printf '%f ')
+  _boards=$(find -L "${BUILD_PATH}/boards" -mindepth 2 -maxdepth 2 -type d -not -path '*/default/*' -printf '%f ')
   complete -W "$_boards" defconfig
   complete -r setconfig 2> /dev/null || return 0
 }
diff --git a/envsetup_soc.sh b/envsetup_soc.sh
index 755eb764..2919cb1d 100755
--- a/envsetup_soc.sh
+++ b/envsetup_soc.sh
@@ -1204,8 +1204,14 @@ function cvi_setup_env()
       FLASH_PARTITION_XML="$BUILD_PATH"/boards/"${CHIP_ARCH,,}"/"$PROJECT_FULLNAME"/partition/partition_"$STORAGE_TYPE".xml
     fi
     if ! [ -e "$FLASH_PARTITION_XML" ]; then
-      print_error "${FLASH_PARTITION_XML} does not exist!!"
-      return 1
+      default_partition_xml="$BUILD_PATH"/partitions.xml
+      if [ -e "$default_partition_xml" ]; then
+        echo "${FLASH_PARTITION_XML} does not exist, default to ${default_partition_xml}"
+        FLASH_PARTITION_XML="$default_partition_xml"
+      else
+        print_error "${FLASH_PARTITION_XML} does not exist!!"
+        return 1
+      fi
     fi
   fi
 
-- 
2.25.1

